x-pack/plugins/ml/public/application/model_management/create_pipeline_for_model/get_inference_properties_from_pipeline_config.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import {
+  IngestInferenceProcessor,
+  IngestInferenceConfig,
+} from '@elastic/elasticsearch/lib/api/types';
+import { isPopulatedObject } from '@kbn/ml-is-populated-object';
+import { SUPPORTED_PYTORCH_TASKS } from '@kbn/ml-trained-models-utils';
+import { DEFAULT_INPUT_FIELD } from '../test_models/models/inference_base';
+
+const INPUT_FIELD = 'inputField';
+const ZERO_SHOT_CLASSIFICATION_PROPERTIES = ['labels', 'multi_label'] as const;
+const QUESTION_ANSWERING_PROPERTIES = ['question'] as const;
+
+const MODEL_INFERENCE_CONFIG_PROPERTIES = {
+  [SUPPORTED_PYTORCH_TASKS.QUESTION_ANSWERING]: QUESTION_ANSWERING_PROPERTIES,
+  [SUPPORTED_PYTORCH_TASKS.ZERO_SHOT_CLASSIFICATION]: ZERO_SHOT_CLASSIFICATION_PROPERTIES,
+} as const;
+
+type SupportedModelInferenceConfigPropertiesType = keyof typeof MODEL_INFERENCE_CONFIG_PROPERTIES;
+
+// Currently, estypes doesn't include pipeline processor types with the trained model processors
+type MLInferencePipelineInferenceConfig = IngestInferenceConfig & {
+  zero_shot_classification?: estypes.MlZeroShotClassificationInferenceOptions;
+  question_answering?: estypes.MlQuestionAnsweringInferenceUpdateOptions;
+};
+
+interface GetInferencePropertiesFromPipelineConfigReturnType {
+  inputField: string;
+  inferenceConfig?: MLInferencePipelineInferenceConfig;
+  inferenceObj?: any;
```

Comment:
It looks like this should be `IngestInferenceProcessor`
Correcting this type highlights some other possibly `undefined` issues where this function is used in `selected_model.tsx` which were hidden by the use of `any`

Reply:
Thanks for catching that `any` that slipped through. Will update but we wouldn't ever get undefined here for the processor since the validation at step one requires at least one inference processor before you can get to this point. 
Agreed that the type should be updated, though. 👍

Reply:
Updated here 546d33e94c81d10a52979c7fb033025a4463226e